src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogSummaryMailGenerator.java

```diff
@@ -102,16 +105,32 @@ public String prepareMailContent(int numberOfDaysBackInThePast, long issueNumber
         ).collect(Collectors.toList());
     }
 
-    private Map<Blog, String> getBlogAndItsHomepage(List<Blog> blogsAddedSinceLastNewsletter) {
-        return blogsAddedSinceLastNewsletter.stream().collect(Collectors.toMap(
-            Function.identity(),
-            blog -> getBlogHomepageFromRss(blog.getRss()))
-        );
+    private Map<Blog, HomePageUrl> getBlogAndItsHomePage(
+        List<Blog> blogsAddedSinceLastNewsletter) {
+        return blogsAddedSinceLastNewsletter.stream()
+            .map(blog -> Pair.of(blog,
```

Comment:
There is definitely something not ok with formatting here (lines 108 - 114). The code is hard to read.